CLI/controllers/update.go

```diff
@@ -16,6 +16,41 @@ func (controller Controller) UpdateObj(pathStr string, data map[string]any, with
 		category = obj["category"].(string)
 	}
 
+	newAttrs, ok := data["attributes"].(map[string]any)
+	if ok {
+		if category == models.EntityToString(models.DEVICE) {
+			currentAttrs := obj["attributes"].(map[string]any)
+			if newAttrs["sizeU"] != nil {
+				sizeU, err := models.CheckSizeUFormat(newAttrs["sizeU"])
+				if err != nil {
+					return nil, err
+				}
+				var height = sizeU * models.RACKUNIT
+				switch heightUnit := currentAttrs["heightUnit"]; heightUnit {
+				case "cm":
+					height *= 100
+				case "mm":
+					height *= 1000
+				}
+				newAttrs["height"] = height
+			}
+			if newAttrs["height"] != nil {
+				height, err := models.CheckHeightFormat(newAttrs["height"])
+				if err != nil {
+					return nil, err
+				}
+				var sizeU = height / models.RACKUNIT
+				switch heightUnit := currentAttrs["heightUnit"]; heightUnit {
+				case "cm":
+					sizeU /= 100
+				case "mm":
+					sizeU /= 1000
+				}
+				newAttrs["sizeU"] = sizeU
+			}
+		}
+	}
+
 	url, err := controller.ObjectUrl(pathStr, 0)
```

Comment:
Should the API check if height and sizeU are coherent and refuse the request if not? In case flutter does not do the same as the CLI, for example.

Reply:
Done in [a2d5002](https://github.com/ditrit/OGrEE-Core/pull/493/commits/a2d500268327f56e85a142eabe8757e74f0ceaca)